File: sdks/typescript/src/resources/admin/api_keys.ts
import type { RequestClient } from "../../core.js";
import type {
  ApiKey,
  ApiKeyListResponse,
  CreateApiKeyBody,
  CreateApiKeyResponse,
  StatusResponse,
  UpdateApiKeyBody,
} from "../../types/index.js";
import { pagination } from "./_shared.js";

export class AdminApiKeysResource {
  constructor(private readonly _client: RequestClient) {}

  list(options: { limit?: number; offset?: number } = {}): Promise<ApiKeyListResponse> {
    return this._client._request("GET", "/v1/admin/api-keys", { params: pagination(options) });
  }

  create(body: CreateApiKeyBody): Promise<CreateApiKeyResponse> {
    return this._client._request("POST", "/v1/admin/api-keys", { json: body });
  }

  update(keyId: string, body: UpdateApiKeyBody): Promise<ApiKey> {
    return this._client._request("PATCH", `/v1/admin/api-keys/${encodeURIComponent(keyId)}`, {
      json: body,
    });
  }

  rotate(keyId: string): Promise<CreateApiKeyResponse> {
    return this._client._request("POST", `/v1/admin/api-keys/${encodeURIComponent(keyId)}/rotate`);
  }

  delete(keyId: string): Promise<StatusResponse> {
    return this._client._request("DELETE", `/v1/admin/api-keys/${encodeURIComponent(keyId)}`);
  }
}

File: sdks/python/src/bigrag/resources/admin/api_keys.py
from __future__ import annotations

from typing import TYPE_CHECKING
from urllib.parse import quote

from bigrag.resources.admin._shared import _pagination
from bigrag.types.admin import (
    ApiKey,
    ApiKeyListResponse,
    CreateApiKeyBody,
    CreateApiKeyResponse,
    UpdateApiKeyBody,
)
from bigrag.types.common import StatusResponse

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminApiKeysResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def list(
        self, *, limit: int | None = None, offset: int | None = None
    ) -> ApiKeyListResponse:
        params = _pagination(limit=limit, offset=offset)
        return await self._client._request("GET", "/v1/admin/api-keys", params=params)

    async def create(self, body: CreateApiKeyBody) -> CreateApiKeyResponse:
        return await self._client._request("POST", "/v1/admin/api-keys", json=body)

    async def update(self, key_id: str, body: UpdateApiKeyBody) -> ApiKey:
        return await self._client._request(
            "PATCH", f"/v1/admin/api-keys/{quote(key_id, safe='')}", json=body
        )

    async def rotate(self, key_id: str) -> CreateApiKeyResponse:
        return await self._client._request(
            "POST", f"/v1/admin/api-keys/{quote(key_id, safe='')}/rotate"
        )

    async def delete(self, key_id: str) -> StatusResponse:
        return await self._client._request("DELETE", f"/v1/admin/api-keys/{quote(key_id, safe='')}")
